Source/com/drew/metadata/mp4/boxes/ItemListBox.java

```diff
@@ -0,0 +1,145 @@
+    {
+        super(box);
+
+        long totalRead = 0;
+        try {
+            while (totalRead < box.size) {
+                long recordLen = reader.getUInt32();
+                String fieldName = reader.getString(4);
+                long fieldLen = reader.getUInt32();
+                String typeName = reader.getString(4);//data
+                totalRead += 16;
+                if ("data".equals(typeName)) {
+                    reader.skip(8);//not sure what these are
+                    totalRead += 8;
+                    int toRead = (int) fieldLen - 16;
+                    if (toRead <= 0) {
+                        //log?
+                        return;
+                    }
+                    if ("covr".equals(fieldName)) {
+                        //covr can be an image file, e.g. png or jpeg
+                        //skip this for now
+                        reader.skip(toRead);
+                    } else if ("cpil".equals(fieldName)) {
+                        int compilationId = (int) reader.getByte();
+//                    metadata.set(XMPDM.COMPILATION, compilationId);
+                    } else if ("trkn".equals(fieldName)) {
+                        if (toRead == 8) {
+                            long numA = reader.getUInt32();
+                            long numB = reader.getUInt32();
+//                        metadata.set(XMPDM.TRACK_NUMBER, (int)numA);
+                        } else {
+                            //log
+                            reader.skip(toRead);
+                        }
+                    } else if ("disk".equals(fieldName)) {
+                        int a = reader.getInt32();
+                        short b = reader.getInt16();
+//                    metadata.set(XMPDM.DISC_NUMBER, a);
+                    } else {
+                        String val = reader.getString(toRead);
+                        if (itemMap.containsKey(fieldName)) {
+                            ItemListEntry entry = itemMap.get(fieldName);
+                            entry.setVal(val);
+                            itemMap.put(fieldName, entry);
+                        }
+//                    try {
+//                        addMetadata(fieldName, val);
+//                    } catch (SAXException e) {
+//                        //need to punch through IOException catching in MP4Reader
+//                        throw new RuntimeSAXException(e);
+//                    }
+                    }
+
+                    totalRead += toRead;
+                } else {
+                    int toSkip = (int) recordLen - 16;
+                    if (toSkip <= 0) {
+                        //log?
+                        return;
+                    }
+                    reader.skip(toSkip);
+                    totalRead += toSkip;
+                }
+            }
+        } catch (EOFException e) {
+            System.out.println("EOF reached");
```

Comment:
The current implementation needs to be modified to avoid EOF.

Apple documents how to properly parse this atom, but upon first glance, it's a bit involved https://developer.apple.com/library/archive/documentation/QuickTime/QTFF/Metadata/Metadata.html

I don't mind leaving this catch here for now.